tests/PHPStan/Analyser/data/bug-3009.php

```diff
@@ -15,14 +15,14 @@ public function createRedirectRequest(string $redirectUri): ?string
 			return null;
 		}
 
-		assertType('array{scheme?: string, host?: string, port?: int<0, 65535>, user?: string, pass?: string, path?: string, query?: string, fragment?: string}', $redirectUrlParts);
```

Comment:
There is a `return null` if `true === array_key_exists('host', $redirectUrlParts)` so the key shouldn't exist anymore.